cmake/OpenCVCompilerOptions.cmake

```diff
@@ -529,3 +529,8 @@ if(CMAKE_GENERATOR MATCHES "Visual Studio" AND CMAKE_CXX_COMPILER_ID MATCHES "MS
     endif()
   endif()
 endif()
+
+if(OPENCV_DUMP_PREPROCESSOR)
```

Comment:
Compiler preprocessor output is a different thing - it usually means `-E` option only and it uses the source files.

This patch `prints macro definitions` as `-dM` option is used.

So, `OPENCV_DUMP_PREPROCESSOR_MACROS` should look better.

Reply:
Updated.